compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -583,6 +576,15 @@ pub(crate) fn global_llvm_features(sess: &Session, diagnostics: bool) -> Vec<Str
         .flatten();
     features.extend(feats);
 
+    // Features implied by an implicit or explicit `--target`.
+    features.extend(
+        sess.target
+            .features
+            .split(',')
+            .filter(|v| !v.is_empty() && backend_feature_name(v).is_some() && v.starts_with("+"))
+            .map(String::from),
+    );
```

Comment:
So this will silently ignore `-Ctarget-features=-x87`? That doesn't seem great, we shouldn't just ignore that. We should hard error or *at least* warn about this.

I think we'll want to compare that in the final feature set, certain features (for x86: x87, soft-float; for other targets we need to still determine this list) have the same state as without applying the user config. This needs to take into account `-Ctarget-cpu` and features implying other features.